src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1241,7 +1241,8 @@ private string GetPreferredExtension(DocumentId documentId, SourceCodeKind sourc
 
         public override IVsHierarchy GetHierarchy(ProjectId projectId)
         {
-            _foregroundObject.AssertIsForeground();
+            // TODO: XAML calls this on a background thread: https://github.com/dotnet/roslyn/issues/36223
+            // _foregroundObject.AssertIsForeground();
 
             // This doesn't take a lock since _projectToGuidMap is immutable
             return _projectToGuidMap.TryGetValue(projectId, out var projectGuid) &&
```

Comment:
We're confident this isn't trading the functional break for a deadlock?

Reply:
No. This is not a proper fix, just workaround.